test/testdata/dsl/t_struct/default_bad.rb

```diff
@@ -2,5 +2,5 @@
 # TODO enable on the fast path
 
 class DefaultBad < T::Struct
-  prop :foo, Integer, default: "bad" # error: Argument does not have asserted type `Integer`
+  prop :foo, Integer, default: "bad" # error: Returning value that does not conform to method result type
```

Comment:
This doesn't seem like a better error message, is there anything that can be done to recover the original?

Reply:
The old way was this: ```$ s test/testdata/dsl/t_struct/default_bad.rb
test/testdata/dsl/t_struct/default_bad.rb:5: Argument does not have asserted type Integer https://srb.help/7007
     5 |  prop :foo, Integer, default: "bad" # error: Returning value that does not conform to method result type
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Got String("bad") originating from:
    test/testdata/dsl/t_struct/default_bad.rb:5:
     5 |  prop :foo, Integer, default: "bad" # error: Returning value that does not conform to method result type
                                       ^^^^^```

The new way is 
```
s test/testdata/dsl/t_struct/default_bad.rb
test/testdata/dsl/t_struct/default_bad.rb:5: Returning value that does not conform to method result type https://srb.help/7005
     5 |  prop :foo, Integer, default: "bad" # error: Returning value that does not conform to method result type
                                       ^^^^^
  Expected Integer
    test/testdata/dsl/t_struct/default_bad.rb:5: Method initialize<defaultArg>1 has return type Integer
     5 |  prop :foo, Integer, default: "bad" # error: Returning value that does not conform to method result type
                                       ^^^^^
  Got String("bad") originating from:
    test/testdata/dsl/t_struct/default_bad.rb:5:
     5 |  prop :foo, Integer, default: "bad" # error: Returning value that does not conform to method result type
                                       ^^^^^
```

Reply:
Maybe I can try to get the loc right for the `Integer` and that will help?

Reply:
I don't like that it talks about a return value in the main content of the error message now. Is there a way that you can get it back to referring to the argument being of the wrong type?

Reply:
Ok, sounds good. I think I have a way to make it go exactly back to the same